Clarify documentation regarding `+bundle` linkage modifier (#112384)
rust-lang/rust: src/doc/rustc/src/command-line-arguments.md

otherwise it will use a static library. If the kind is specified on the
command-line, it will override the kind specified in a `link` attribute.

> Note that if you don't specify the `KIND`, the default linkage modifiers won't be applied.

Reviewer: What does "default linkage modifiers" mean?
There are no default linkage modifiers.

Author: I meant this. It's enabled by default for `static` kind https://github.com/rust-lang/rust/blob/46c5ee11bb47de0cba0b4538537e1e3813088583/src/doc/rustc/src/command-line-arguments.md?plain=1#L109

Reviewer 2: ```suggestion
If you don't specify `KIND`, no linkage modifiers will be applied.
```

This might be a slightly clearer wording..? If I understand correctly, when `KIND` is not specified then the `+` or `-` default for the modifiers doesn't apply.

> For instance, [`+bundle` linkage modifier][bundle-modifier] won't be used in conjunction with static library by default.

Reviewer: Of course it won't be used, because there won't be any static library.
When this PR was submitted, unspecified kind always meant `dylib`.

> For instance, [`+bundled` linkage modifier][bundled-modifier] won't be used in conjunction with `static` by default.

Reviewer: ```suggestion
For instance, [`+bundle` linkage modifier][bundled-modifier] won't be used in conjunction with `static` by default.
```

> For instance, [`+bundled` linkage modifier][bundled-modifier] won't be used in conjunction with `static` by default.

Reviewer: If there's no `KIND`, then there's no `static`, I'm not sure what this paragraph is trying to clarify.

Reviewer 2: This might be a little clearer?

```
For example, the `bundled` linking modifier's default is `+bundled`, which would not be applied to a static library (where it otherwise would have been).
```
